src/setfit/modeling.py

```diff
@@ -404,7 +425,7 @@ def predict_proba(self, x_test: List[str], as_numpy: bool = False) -> Union[torc
         outputs = self.model_head.predict_proba(embeddings)
 
         if as_numpy and self.has_differentiable_head:
-            outputs = outputs.cpu().numpy()
+            outputs = outputs.cpu().detach().numpy()
```

Comment:
Good call! I believe `outputs.detach().cpu().numpy()` is more efficient, so I'll go with that. I'll also apply it for `predict`.